indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -83,6 +92,47 @@ protected void map(
     }
   }
 
-  abstract protected void innerMap(InputRow inputRow, Text text, Context context)
+  public final static InputRow parseInputRow(Writable value, InputRowParser parser)
+  {
+    if(value instanceof Text) {
+      //Note: this hack is required because StringInputRowParser is a
+      //InputRowParser<ByteBuffer> and not InputRowParser<String>
+      //parse() method on it expects a ByteBuffer and not String
+      ByteBuffer bb = ByteBuffer.wrap(value.toString().getBytes());
```

Comment:
@gianm @drcrallen 
actually changed code to look like
"((StringInputRowParser)parser).parse(value.toString());"
It is like old code and wouldn't need byte buffer wrapping business.